src/game/server/ddracechat.cpp

```diff
@@ -1567,21 +1578,105 @@ void CGameContext::ConTele(IConsole::IResult *pResult, void *pUserData)
 		}
 		if(ClientID == MAX_CLIENTS)
 		{
-			pSelf->SendChatTarget(pPlayer->GetCID(), "No player with this name found.");
+			pSelf->SendChatTarget(pCallingPlayer->GetCID(), "No player with this name found.");
 			return;
 		}
-		CPlayer *pPlayerTo = pSelf->m_apPlayers[ClientID];
-		if(!pPlayerTo)
+
+		CPlayer *pDestPlayer = pSelf->m_apPlayers[ClientID];
+		if(!pDestPlayer)
 			return;
-		CCharacter *pChrTo = pPlayerTo->GetCharacter();
-		if(!pChrTo)
+		CCharacter *pDestCharacter = pDestPlayer->GetCharacter();
+		if(!pDestCharacter)
 			return;
-		Pos = pChrTo->m_Pos;
+
+		// Set calling tee's position to that of the destination tee
+		Pos = pDestCharacter->m_Pos;
+		break;
 	}
-	pSelf->Teleport(pChr, Pos);
-	pChr->UnFreeze();
-	pChr->Core()->m_Vel = vec2(0, 0);
-	pPlayer->m_LastTeleTee.Save(pChr);
+	case 2:
+	{
+		float BaseX = 0.f, BaseY = 0.f;
+
+		CMapItemLayerTilemap *pGameLayer = pSelf->m_Layers.GameLayer();
+		constexpr const float OuterKillTileBoundaryDistance = 201 * 32.f;
+		float MapWidth = (pGameLayer->m_Width * 32) + (OuterKillTileBoundaryDistance * 2.f), MapHeight = (pGameLayer->m_Height * 32) + (OuterKillTileBoundaryDistance * 2.f);
+
+		const auto DetermineCoordinateRelativity = [](const char *InString, float AbsoluteDefaultValue, float &OutFloat) -> bool {
+			// mode 0 = abs, 1 = sub, 2 = add
+			uint8_t Mode;
+			const char *pStrDelta = str_startswith(InString, "~");
+			if(!pStrDelta || str_startswith(pStrDelta, "-"))
+				pStrDelta = str_startswith(InString, "~+");
+
+			// Doesn't start with ~ or ~+ so we are not in relative add mode
+			if(!pStrDelta)
+			{
+				// Doesn't start with - so we're not in relative subtract mode
+				pStrDelta = str_startswith(InString, "~-");
+				if(!pStrDelta)
+				{
+					// Starts with neither so the string is either invalid or we're in absolute mode
+					Mode = 0;
+					pStrDelta = InString;
+				}
+				else // Relative subtract mode
+					Mode = 1;
+			}
+			else // Relative add mode
+				Mode = 2;
+
+			// Is the number valid?
+			float d = std::atof(pStrDelta);
+			if(std::isnan(d) || std::isinf(d) || d == HUGE_VALF)
```

Comment:
```suggestion
			if(std::isnan(d) || std::isinf(d))
```

`HUGE_VALF` seems to be infinity.